src/hooks/useAppOpenAd.ts

```diff
@@ -37,7 +37,8 @@ export function useAppOpenAd(
   const [appOpenAd, setAppOpenAd] = useState<AppOpenAd | null>(null);
 
   useDeepCompareEffect(() => {
-    setAppOpenAd(() => {
+    setAppOpenAd(prevAd => {
```

Comment:
if we null-check the type and it's not required, shouldn't the type of the setAppOpenAd function parameter be nullable? 🤔 

```suggestion
    setAppOpenAd(prevAd? => {
```

Reply:
The parameter becomes automatically nullable because type of `appOpenAd` state is `AppOpenAd | null`?